lib/linked_list.rb

```diff
@@ -7,39 +8,72 @@ def initialize
     @head = nil
   end
 
-  # Time complexity - ?
-  # Space complexity - ?
+  # Time complexity - O(1) where 1 is the one node created
+  # Space complexity - O(1) where 1 is the one node created in space
   def add_first(data)
-
+    @head = Node.new(data, head)
   end
 
-  # Time complexity - ?
-  # Space complexity - ?
+  # Time complexity - O(1) because it is grabbing the first node in the list
+  # Space complexity - O(1) because it is returning one node
   def get_first
-
+    return nil if head.nil?
+    return @head.data
   end
 
-  # Time complexity - ?
-  # Space complexity - ?
+  # Time complexity - O(n) where n is the length of the linked list
+  # Space complexity - O(1) because only a single number is being returning, the length of the list
   def length
-    return 0
+    count = 0
+    current = @head
+    until current.nil?
+      count += 1
+      current = current.next
+    end
+    return count
   end
 
-  # Time complexity - ?
-  # Space complexity - ?
+  # Time complexity - O(n) where n is the length of the entire list
+  # Space complexity - O(1) because you are adding one node
   def add_last(data)
+    current = @head
 
+    if current.nil?
+      add_first(data)
+      return
+    end
+
+    until current.next == nil
+      current = current.next
+    end
+    current.next = Node.new(data)
   end
 
-  # Time complexity - ?
-  # Space complexity - ?
+  # Time complexity - O(n) where n is the length of the entire list
+  # Space complexity - O(1) because you are returning informatio on one node
   def get_last
+    current = @head
```

Comment:
What if `head` is `nil`?

```suggestion
    return nil if head.nil?
    current = head
```